### ISSUE TITLE
Have a problem with build multilayer perceptron

### ISSUE THREAD

**Issue by inferrna** [raskr/rust-autograd#31]
Hi. I tried to build simple perceptron with more than 1 hidden layer, but its precision seems to be very low. First 10 epochs it shows some progress, but then stuck on error ~ 0.585, which is very large for task detecting value from 0.0 to 1.0.
Can you please look on it and say what I did wrong?
Here is it https://gist.github.com/inferrna/b1b49b9d0e161a5104670ecb7870e19a

**Comment by raskr** [comment]
I'm sorry I can't figure out the reason why that is, but 
```rust
let values = z.eval(&[x.given(x_test.view()), y.given(y_test.view())]).unwrap();
```
should be like below because `z` is not normalized to be between 0~1.
```rust
let values = g.sigmoid(z).eval(&[x.given(x_test.view()), y.given(y_test.view())]).unwrap();
```


> stuck on error ~ 0.585, which is very large for task detecting value from 0.0 to 1.0.

In your code, `acc` seems to be (binary) cross entropy. What is the desired value in the dataset?

**Comment by inferrna** [comment]
> What is the desired value in the dataset?

I try to predict float value by its cosine and sine. More precisely, input is `[(sin(y)+1.0)/2.0, (cos(y)+1.0)/2.0]`

I changed (also in gist)
```rust
let z = g.matmul(xww2, w3) + b;
```
to
```rust
let z = g.sigmoid(g.matmul(xww2, w3) + b);
```
and
```rust
let mean_loss = g.reduce_mean(g.sigmoid_cross_entropy(z, &y), &[0,1], false);
```
to
```rust
let mean_loss = g.reduce_mean(g.square(g.sub(z, &y)), &[0,1], false);
```
Now error is near 0.03 which is much better (but still not perfect). Thank you for help.
(Although, it would be very nice to have example for such cases with MSE loss function and a lot of layers and etc - for those ones who migrating from fann)

**Comment by raskr** [comment]
> Although, it would be very nice to have example for such cases with MSE loss function and a lot of layers and etc - for those ones who migrating from fann)

Sorry about the poor example code.
I know that the num of use cases is important for this kind of tools, but I don't have enough time to do it...
